modules/reserved-ips/variables.tf

```diff
@@ -0,0 +1,126 @@
+      cidr = optional(string)
+    })
+  )
+  default = []
+}
+
+
+##############################################################################
+# VPE Variables
+##############################################################################
+
+variable "region" {
+  description = "The region where VPC and services are deployed"
+  type        = string
+  default     = "us-south"
+}
+
+variable "cloud_services" {
+  description = "List of cloud services to create an endpoint gateway. The keys are the service names, the values (all optional) give some level of control on the created VPEs."
+  type = set(object({
+    service_name                 = string
+    vpe_name                     = optional(string), # Full control on the VPE name. If not specified, the VPE name will be computed based on prefix, vpc name and service name.
+    allow_dns_resolution_binding = optional(bool, false)
+  }))
+  default = []
+  validation {
+    error_message = "Currently the service you're trying to add is not supported. Any other VPE services must be added using `cloud_service_by_crn`."
+    condition = length(var.cloud_services) == 0 ? true : length([
+      for service in var.cloud_services :
+      service.service_name if !contains([
+        "account-management",
+        "billing",
+        "cloud-object-storage",
+        "cloud-object-storage-config",
+        "codeengine",
+        "container-registry",
+        "containers-kubernetes",
+        "context-based-restrictions",
+        "directlink",
+        "dns-svcs",
+        "enterprise",
+        "global-search-tagging",
+        "globalcatalog",
+        "hs-crypto",
+        "hs-crypto-cert-mgr",
+        "hs-crypto-ep11",
+        "hs-crypto-ep11-az1",
+        "hs-crypto-ep11-az2",
+        "hs-crypto-ep11-az3",
+        "hs-crypto-kmip",
+        "hs-crypto-tke",
+        "hyperp-dbaas-mongodb",
+        "hyperp-dbaas-postgresql",
+        "iam-svcs",
+        "is",
+        "kms",
+        "messaging",
+        "resource-controller",
+        "support-center",
+        "transit",
+        "user-management",
+        "vmware",
+        "ntp"
+      ], service.service_name)
+    ]) == 0
+  }
+}
+
+variable "cloud_service_by_crn" {
+  description = "List of cloud service CRNs. Each CRN will have a unique endpoint gateways created. For a list of supported services, see the docs [here](https://cloud.ibm.com/docs/vpc?topic=vpc-vpe-supported-services)."
+  type = list(
+    object({
+      name = string # service name
+      crn  = string # service crn
+    })
+  )
+  default = []
+}
+
+variable "endpoint_ip_list" {
+  description = "List of IPs to create. Each object contains an IP name and subnet id"
+  type = list(
+    object({
+      ip_name      = string # reserved ip name
+      subnet_id    = string # subnet id
+      gateway_name = string # gateway name
+      name         = string # ip name
+    })
+  )
+  default = []
+}
+
+variable "reserved_ips" {
+  description = "Reserved IPs"
```

Comment:
Perhaps give a more descriptive description here. Something like "Map of reserved IP name and values that will be created by this module."